src/main/java/org/checkerframework/specimin/JavaParserUtil.java

```diff
@@ -64,6 +76,35 @@ private JavaParserUtil() {
     throw new UnsupportedOperationException("This class cannot be instantiated.");
   }
 
+  /**
+   * Set this TypeSolver instance to be used by the JavaParserFacade, so we don't have to find it
+   * through reflection in JavaParserSymbolSolver. This field is initialized once in SpeciminRunner.
+   * Note that by the time you evaluate the value of this field, it should already be non-null.
```

Comment:
The [MonotonicNonNull annotation](https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/MonotonicNonNull.html) exactly expresses this contract.

Reply:
(you'll also want an `@EnsuresNonNull` annotation on `setTypeSolver`, I think)